test/test_mps.py

```diff
@@ -1841,6 +1841,17 @@ def test_slice_reshape_contg_view(self):
 
         self.assertEqual(r_mps, r_cpu)
 
+    def test_contiguous_slice(self):
+        t_mps = torch.randn([3, 4], device="mps")
+        t_cpu = t_mps.detach().clone().cpu()
+
```

Comment:
Can you add the test case where the storage offset is slicing 3D , 4D tensor with slice offset ?